packages/vite-plugin-cloudflare/src/build.ts

```diff
@@ -0,0 +1,176 @@
+	resolvedPluginConfig: ResolvedPluginConfig
+): (builder: vite.ViteBuilder) => Promise<void> {
+	return async (builder) => {
+		const clientEnvironment = builder.environments.client;
+		assert(clientEnvironment, `No "client" environment`);
+		const defaultHtmlPath = path.resolve(builder.config.root, "index.html");
+		const hasClientEntry =
+			clientEnvironment.config.build.rollupOptions.input ||
+			fs.existsSync(defaultHtmlPath);
+
+		if (resolvedPluginConfig.type === "assets-only") {
+			if (hasClientEntry) {
+				await builder.build(clientEnvironment);
+			} else if (getHasPublicAssets(builder.config)) {
+				await fallbackBuild(builder, clientEnvironment);
+			}
+
+			// Return early as there are no Workers to build
+			return;
+		}
+
+		const workerEnvironments = Object.keys(resolvedPluginConfig.workers).map(
+			(environmentName) => {
+				const environment = builder.environments[environmentName];
+				assert(environment, `"${environmentName}" environment not found`);
+
+				return environment;
+			}
+		);
+
+		await Promise.all(
+			workerEnvironments.map((environment) => builder.build(environment))
+		);
+
+		const { entryWorkerEnvironmentName } = resolvedPluginConfig;
+		const entryWorkerEnvironment =
+			builder.environments[entryWorkerEnvironmentName];
+		assert(
+			entryWorkerEnvironment,
+			`No "${entryWorkerEnvironmentName}" environment`
+		);
+		const entryWorkerBuildDirectory = path.resolve(
+			builder.config.root,
+			entryWorkerEnvironment.config.build.outDir
+		);
+		const entryWorkerManifest = loadViteManifest(entryWorkerBuildDirectory);
+		const importedAssetPaths = getImportedAssetPaths(entryWorkerManifest);
+
+		if (hasClientEntry) {
+			await builder.build(clientEnvironment);
+		} else if (importedAssetPaths.size || getHasPublicAssets(builder.config)) {
+			await fallbackBuild(builder, clientEnvironment);
+		} else {
+			const entryWorkerConfigPath = path.join(
+				entryWorkerBuildDirectory,
+				"wrangler.json"
+			);
+			const workerConfig = JSON.parse(
+				fs.readFileSync(entryWorkerConfigPath, "utf-8")
+			) as Unstable_Config;
+			// Remove `assets` field as there are no assets
+			workerConfig.assets = undefined;
+			fs.writeFileSync(entryWorkerConfigPath, JSON.stringify(workerConfig));
+
+			// Return early as there is no client build
+			return;
+		}
+
+		const clientBuildDirectory = path.resolve(
+			builder.config.root,
+			clientEnvironment.config.build.outDir
+		);
+		const movedAssetPaths: string[] = [];
+
+		// Move assets imported in the entry Worker to the client build
+		for (const assetPath of importedAssetPaths) {
+			const src = path.join(entryWorkerBuildDirectory, assetPath);
+			const dest = path.join(clientBuildDirectory, assetPath);
+
+			if (!fs.existsSync(src)) {
+				continue;
+			}
+
+			if (fs.existsSync(dest)) {
+				fs.unlinkSync(src);
+			} else {
+				const destDir = path.dirname(dest);
+
+				if (!fs.existsSync(destDir)) {
+					fs.mkdirSync(destDir, { recursive: true });
+				}
```

Comment:
It might well just be quicker to do?

```suggestion
				fs.mkdirSync(destDir, { recursive: true });
```

Reply:
[365c4da](https://github.com/cloudflare/workers-sdk/pull/9510/commits/365c4daecfebe68125e645c7fde229cc4b4129e9)